### PR TITLE
Feature/36584 pupil aliases

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
Pupils Overview:

- [x] Display total number of pupils on UI

Add/Edit pupil:
- [x] Added pupil first and last name aliases on UI
- [x] Submit aliases

### CODE REVIEW

File: admin/data/sql/migrations/20191112222703.do.add-aliases-pupil-table.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE mtc_admin.pupil ADD foreNameAlias [NVARCHAR](max) NULL
```
Comment (jon-shipley) [inline admin/data/sql/migrations/20191112222703.do.add-aliases-pupil-table.sql]: MAX can be 2GB which seems a little high for an alias.  Perhaps both of these aliases should be limited to 128 chars?  How much of the 2GB would we actually display?

@GuyHarwood - thoughts?

File: admin/data/sql/migrations/20191112222703.undo.add-aliases-pupil-table.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE mtc_admin.pupil DROP COLUMN lastNameAlias
```
Comment (jon-shipley) [inline admin/data/sql/migrations/20191112222703.undo.add-aliases-pupil-table.sql]: IMO, SQL should be terminated by a semi-colon.

File: admin/data/sql/migrations/20191112222703.do.add-aliases-pupil-table.sql (hunk 1 above)
Comment (GeorgeChatzigiannis) [reply admin/data/sql/migrations/20191112222703.do.add-aliases-pupil-table.sql]: @jon-shipley we should change then middleNames as well as this follows the same pattern

Comment (jon-shipley) [reply admin/data/sql/migrations/20191112222703.do.add-aliases-pupil-table.sql]: Good idea.

Comment (GuyHarwood) [reply admin/data/sql/migrations/20191112222703.do.add-aliases-pupil-table.sql]: agreed

File: admin/data/sql/migrations/20191112222703.undo.add-aliases-pupil-table.sql (hunk 2 above)
Comment (GeorgeChatzigiannis) [reply admin/data/sql/migrations/20191112222703.undo.add-aliases-pupil-table.sql]: done

File: admin/data/sql/migrations/20191112222703.do.add-aliases-pupil-table.sql (hunk 1 above)
Comment (GeorgeChatzigiannis) [reply admin/data/sql/migrations/20191112222703.do.add-aliases-pupil-table.sql]: done

**Comment by GeorgeChatzigiannis** [comment]
@GuyHarwood @jon-shipley thanks for the comments, need another review on the migration changes please.

File: admin/data/sql/migrations/20191112222703.do.add-aliases-pupil-table.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
+DROP INDEX [mtc_admin].[pupil].[idx_azure_recommended_pupil_school];
+ALTER TABLE mtc_admin.pupil ALTER COLUMN foreName [NVARCHAR](128) NULL;
+ALTER TABLE mtc_admin.pupil ALTER COLUMN lastName [NVARCHAR](128) NULL;
+ALTER TABLE mtc_admin.pupil ALTER COLUMN middleNames [NVARCHAR](128) NULL;
+CREATE NONCLUSTERED INDEX [idx_azure_recommended_pupil_school] ON [mtc_admin].[pupil] ([school_id]) INCLUDE ([createdAt], [dateOfBirth], [foreName], [gender], [isTestAccount], [job_id], [jwtSecret], [jwtToken], [lastName], [middleNames], [pin], [pinExpiresAt], [pupilAgeReason_id], [pupilStatus_id], [updatedAt], [upn], [urlSlug], [version]) WITH (ONLINE = ON)
```
Comment (jon-shipley) [inline admin/data/sql/migrations/20191112222703.do.add-aliases-pupil-table.sql]: You may as well include the new fields too.

Comment (GeorgeChatzigiannis) [reply admin/data/sql/migrations/20191112222703.do.add-aliases-pupil-table.sql]: good catch

Comment (GeorgeChatzigiannis) [reply admin/data/sql/migrations/20191112222703.do.add-aliases-pupil-table.sql]: done

**Pull request merged**